executor/new_distsql.go

```diff
@@ -353,64 +358,45 @@ func (ih *indexWorker) fetchHandles(e *IndexLookUpExecutor, result distsql.Selec
 			case <-finished:
 				return
 			case workCh <- task:
+				e.resultCh <- task
```

Comment:
I think there need another `select`, if `resultCh` is full, and `IndexLookupExecutor` stopped reading then call `Close`, indexWorker may be blocked here forever.

But it only need to select `finished`, `ctx.Done` can be handled in the next loop.

The old implementation before and after the refactor both have this issue, it's very hard to trigger though.

Reply:
`TestIndexDoubleReadClose` actually cover the case and that's why my first commit fail to pass CI.
I've fix it by drain the `resultCh` in `IndexLookUpExecutor.Close` https://github.com/pingcap/tidb/pull/4470/files#diff-f77833966b29c503e5221562eb37f86fR552

Reply:
ok